Tests/FinalTest/GradeApp/Data/Grade.cs

```diff
@@ -0,0 +1,18 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace GradeApp.Data;
+
+public class Grade
+{
+    public int GradeId { get; set; }
+
+    [Required(ErrorMessage = "Please enter student name")]
+    public string Name { get; set; } = "";
+
+    [Required(ErrorMessage = "Please enter subject")]
+    public string Subject { get; set; } = "";
+
+    [Required(ErrorMessage = "Please enter grade value")]
+    [Range(0, 10, ErrorMessage = "Please enter a valid grade value. From 0 to 10.")]
+    public string Value { get; set; } = "";
```

Comment:
```suggestion
    public int? Value { get; set; } = "";
```
Вот так и Required будет работать в случае пустой строки на входе, и в БД будет int храниться, а не text